### PR TITLE
[VM -> Broker] [Pytest] Conversion for cli/test_katello_agent

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
**Description:**
1. Refactor for Pytest Conversion.
2. VM -> Broker Conversion for content hosts.
3. Bump tools and capsule repo versions to 6.9.

### CODE REVIEW

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_katello_agent.py

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

File: tests/foreman/cli/test_katello_agent.py [hunk 1]
```diff
@@ -19,20 +19,16 @@
 import time
 
 import pytest
+from broker.broker import VMBroker
```
[the remark this answers is not in the document]
Comment (Gauravtalreja1) [reply tests/foreman/cli/test_katello_agent.py]: Done

File: tests/foreman/cli/test_katello_agent.py [hunk 2]
```diff
@@ -46,288 +42,291 @@
...
-            hosts = HostCollection.hosts({'id': hc['id'], 'organization-id': self.org['id']})
-            assert len(hosts) == 2
-            expected_hosts_ids = {self.host['id'], client_host['id']}
-            hosts_ids = {host['id'] for host in hosts}
-            assert hosts_ids == expected_hosts_ids
+from robottelo.hosts import ContentHost
+
+
+@skip_if_not_set('clients', 'fake_manifest')
+@pytest.fixture(scope="module")
```
[the remark this answers is not in the document]
Comment (Gauravtalreja1) [reply tests/foreman/cli/test_katello_agent.py]: Done.

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_katello_agent.py

**Review by JacobCallahan — APPROVED** [review]
ACK, nicely done!

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_katello_agent.py

**Comment by devendra104** [comment]
@Gauravtalreja1 Please upload the test result as robottelo-runner is failed.

File: tests/foreman/cli/test_katello_agent.py [hunk 3]
```diff
@@ -46,288 +42,291 @@
...
+    return {
+        'ak': module_ak,
+        'cv': module_cv,
+        'lce': module_lce,
+        'org': module_org,
+    }
+
+
+@skip_if_not_set('clients')
+@pytest.fixture(scope='function')
```
[the remark this answers is not in the document]
Comment (Gauravtalreja1) [reply tests/foreman/cli/test_katello_agent.py]: Good catch! 👍

File: tests/foreman/cli/test_katello_agent.py [hunk 4]
```diff
@@ -46,288 +42,291 @@
...
+
+@skip_if_not_set('clients', 'fake_manifest')
+@pytest.fixture(scope='module')
+def katello_agent_repos(module_ak, module_cv, module_lce, module_org):
+    """Create Org, Lifecycle Environment, Content View, Activation key"""
+    setup_org_for_a_rh_repo(
+        {
+            'product': PRDS['rhel'],
+            'repository-set': REPOSET['rhst7'],
+            'repository': REPOS['rhst7']['name'],
```
[the remark this answers is not in the document]
Comment (Gauravtalreja1) [reply tests/foreman/cli/test_katello_agent.py:56]: @devendra104 I didn’t realize it uses 6.8 repos earlier, included a new commit to bump satellite tools, and capsule repos to 6.9 ... Thank you!

**Comment by Gauravtalreja1** [comment]
Included a new [commit](https://github.com/SatelliteQE/robottelo/pull/8556/commits/129ed38f3c3baecf866c7046787298a51e7e0ac2) to bump satellite tools, and capsule repos versions to 6.9 

Since PRT is failing due to connection/VPN issues with AT, so running these tests locally,
**Test Results:**
```
(.sat_env) [gtalreja@gtalreja robottelo]$ pytest -xv tests/foreman/cli/test_katello_agent.py
========================================================== test session starts ==========================================================
platform linux -- Python 3.8.7, pytest-6.2.3, py-1.9.0, pluggy-0.13.1 -- /home/gtalreja/sat_workspace/robottelo/.sat_env/bin/python3
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/gtalreja/sat_workspace/robottelo, configfile: pyproject.toml
plugins: cov-2.11.1, reportportal-5.0.8, xdist-2.2.1, ibutsu-1.14.1, mock-3.5.1, services-2.2.1, forked-1.3.0
collected 10 items

tests/foreman/cli/test_katello_agent.py::test_positive_get_errata_info PASSED                                                     [ 10%]
tests/foreman/cli/test_katello_agent.py::test_positive_apply_errata PASSED                                                        [ 20%]
tests/foreman/cli/test_katello_agent.py::test_positive_apply_security_erratum PASSED                                              [ 30%]
tests/foreman/cli/test_katello_agent.py::test_positive_install_package PASSED                                                     [ 40%]
tests/foreman/cli/test_katello_agent.py::test_positive_remove_package PASSED                                                      [ 50%]
tests/foreman/cli/test_katello_agent.py::test_positive_upgrade_package PASSED                                                     [ 60%]
tests/foreman/cli/test_katello_agent.py::test_positive_upgrade_packages_all PASSED                                                [ 70%]
tests/foreman/cli/test_katello_agent.py::test_positive_install_and_remove_package_group PASSED                                    [ 80%]
tests/foreman/cli/test_katello_agent.py::test_negative_unregister_and_pull_content PASSED                                         [ 90%]
tests/foreman/cli/test_katello_agent.py::test_positive_register_host_ak_with_host_collection PASSED                               [100%]

==================================================== 10 passed in 4301.61s (1:11:41) ====================================================
```

**Review by JacobCallahan — APPROVED** [review]

**Comment by swadeley** [comment]
For future reference, bumping versions for tools etc. is best done in separate git commits as it makes it easier for others to cherry-pick.

File: robottelo/hosts.py [hunk 5]
```diff
@@ -62,7 +62,7 @@ def download_install_rpm(self, repo_url, package_name):
         :raises robottelo.hosts.ContentHostError: If package wasn't installed.
 
         """
-        self.execute(f'wget -nd -r -l1 --no-parent -A \'{package_name}.rpm\' {repo_url}')
+        self.execute(f'curl -O {repo_url}/{package_name}.rpm')
```
Comment (swadeley) [inline robottelo/hosts.py:65]: that is an interesting and welcome change if it works, as curl is installed by default and wget is not. I wonder did @JacobCallahan make a note of that?

**Review by devendra104 — APPROVED** [review]
ACK

File: robottelo/hosts.py (hunk 5 above)
Comment (JacobCallahan) [reply robottelo/hosts.py:65]: @swadeley can you make this a blocking request?

File: tests/foreman/cli/test_katello_agent.py [hunk 6]
```diff
@@ -46,288 +42,291 @@
...
+    setup_org_for_a_custom_repo(
+        {
+            'url': FAKE_1_YUM_REPO,
+            'organization-id': module_org.id,
+            'content-view-id': module_cv.id,
+            'lifecycle-environment-id': module_lce.id,
+            'activationkey-id': module_ak.id,
+        }
+    )
+    return {
```
Comment (mshriver) [inline tests/foreman/cli/test_katello_agent.py:73]: This isn't a required change, but I'm personally not a fan of this approach.

The tests that need these entities can simply include `module_ak`, `module_cv`, etc in their fixtures list, and reference them directly. Doing it this way turns those direct references into dictionary references dependent on the keys set here.

File: tests/foreman/cli/test_katello_agent.py [hunk 7]
```diff
@@ -46,288 +42,291 @@
...
+    # Register content host and install katello-agent
+    rhel7_contenthost.register_contenthost(
+        katello_agent_repos['org'].label,
+        katello_agent_repos['ak'].name,
+    )
+    assert rhel7_contenthost.subscribed
+    host_info = Host.info({'name': rhel7_contenthost.hostname})
+    rhel7_contenthost.enable_repo(REPOS['rhst7']['id'])
+    rhel7_contenthost.install_katello_agent()
+    yield {'client': rhel7_contenthost, 'host_info': host_info}
```
Comment (mshriver) [inline tests/foreman/cli/test_katello_agent.py:94]: Same here, this fixture can just return `host_info`, and tests could reference `rhel7_contenthost` as provided through its own fixture.

**Review by mshriver — APPROVED** [review]

**Pull request merged**